src/fdb5/tools/fdb-move-mpi.cc

```diff
@@ -0,0 +1,197 @@
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+#include "eckit/mpi/Comm.h"
+#include "eckit/option/CmdArgs.h"
+#include "eckit/option/SimpleOption.h"
+
+#include "fdb5/tools/FDBVisitTool.h"
+#include "fdb5/api/FDB.h"
+#include "fdb5/api/helpers/FDBToolRequest.h"
+#include "fdb5/LibFdb5.h"
+#include "fdb5/toc/TocCommon.h"
+
+#define MAX_THREADS 256
+
+using namespace eckit::option;
+using namespace eckit;
+
+namespace fdb5 {
+namespace tools {
+
+//----------------------------------------------------------------------------------------------------------------------
+
+class FDBMove : public FDBVisitTool {
+public: // methods
+
+    FDBMove(int argc, char **argv);
+    ~FDBMove() override;
+
+private: // methods
+
+    void execute(const CmdArgs& args) override;
+    void init(const CmdArgs &args) override;
+
+private: // members
+
+    eckit::URI destination_;
+    bool keep_;
+    int removeDelay_;
+    int threads_;
+    bool mpi_;
+    int rank_;
+};
+
+FDBMove::FDBMove(int argc, char **argv) :
+    FDBVisitTool(argc, argv, "class,expver,stream,date,time"),
+    destination_(""),
+    keep_(false),
+    removeDelay_(0),
+    threads_(1),
+    mpi_(false),
+    rank_(-1) {
+
+    options_.push_back(new SimpleOption<std::string>("dest", "Destination root"));
+    options_.push_back(new SimpleOption<bool>("keep", "Keep source DB"));
+    options_.push_back(new SimpleOption<long>("delay", "Delay in seconds before deleting source (default: 0)"));
+    options_.push_back(new SimpleOption<bool>("mpi", "distributed data move (MPI based)"));
+    options_.push_back(new SimpleOption<long>("threads", "Number of concurrent threads for data move (default: 1)"));
+}
+
+FDBMove::~FDBMove() {}
+
+
+void FDBMove::init(const CmdArgs& args) {
+
+    FDBVisitTool::init(args);
+
+    keep_ = args.getBool("keep", false);
+    mpi_ = args.getBool("mpi", false);
+    removeDelay_ = args.getInt("delay", 0);
+
+    eckit::mpi::Comm& comm = eckit::mpi::comm();
+
+    if (mpi_) {
+        threads_ = 1;
+        rank_ = int(comm.rank());
+    } else {
+        threads_ = args.getInt("threads", 1);
+    }
+
+    if (!mpi_ || rank_ == 0) {
+        std::string dest = args.getString("dest");
+        if (dest.empty()) {
+            std::stringstream ss;
+            ss << "No destination root specified.";
+            throw UserError(ss.str(), Here());
+        } else {
+            destination_ = eckit::URI(dest);
+        }
+
+        if (mpi_ || (0 < threads_ && threads_ <= MAX_THREADS)) {
+            destination_ = eckit::URI(dest);
+        } else {
+            std::stringstream ss;
+            ss << "Unsupported number of threads. please specify a value between 1 and " << MAX_THREADS;
+            throw UserError(ss.str(), Here());
+        }
+    } else { // this process is just a data mover - subscribing to receive tasks
+        char task[1024];
```

Comment:
why 1024? It is perfectly reasonable to have two paths that are longer than that in aggregate. Unlikely, perhaps, but by no means impossible.

Is this something that we should be constructing as, say 2*(PATH_MAX+1)?